tests/test_server.py

```diff
@@ -181,6 +181,54 @@ async def test_autostart_batched(client, server, environment):
     assert len(sessionendpoint._sessions) == 1
 
 
+@pytest.mark.parametrize(
+    "n_versions_to_keep, n_versions_to_create",
+    [
+        (2, 4),
+        (4, 2),
+        (2, 2),
+    ],
+)
+async def test_create_too_many_versions(client, server, n_versions_to_keep, n_versions_to_create):
+    """
+    - set AVAILABLE_VERSIONS_TO_KEEP environment setting to <n_versions_to_keep>
+    - create <n_versions_to_create> versions
+    - check the actual number of versions before and after cleanup
+    """
+
+    # Create project
+    result = await client.create_project("env-test")
+    assert result.code == 200
+    project_id = result.result["project"]["id"]
+
+    # Create environment
+    result = await client.create_environment(project_id=project_id, name="env_1")
+    env_1_id = result.result["environment"]["id"]
```

Comment:
I think it would be useful to check the different setting values with multiple environments in the same test case, because the global config option changed to an environment specific setting. If it was still a global option the test could still pass.

Reply:
I see ! Good point